File: solutions/go/resistor-color-trio/1/resistor_color_trio.go
package resistorcolortrio

import ( "fmt" )
import ( "math" )

// ColorCode returns the resistance value of the given color.
func ColorCode(color string) int {
    colors := []string{
		"black",
		"brown",
		"red",
		"orange",
		"yellow",
		"green",
		"blue",
		"violet",
		"grey",
		"white",
	}
    for index := range colors {
        if colors[index] == color {
            return index
        }
    }
    return -1
}

// Label describes the resistance value given the colors of a resistor.
// The label is a string with a resistance value with an unit appended
// (e.g. "33 ohms", "470 kiloohms").
func Label(colors []string) string {
    resistanceValue := (10 * ColorCode(colors[0]) + ColorCode(colors[1])) * int(math.Pow(10, float64(ColorCode(colors[2]))))
    unit := "ohms"
    if resistanceValue >= 1000000000 {
        resistanceValue /= 1000000000
        unit = "gigaohms"
    } else if resistanceValue >= 1000000 {
        resistanceValue /= 1000000
        unit = "megaohms"
    } else if resistanceValue >= 1000 {
        resistanceValue /= 1000
        unit = "kiloohms"
    } 
    return fmt.Sprintf("%d %s", resistanceValue, unit)
}

File: solutions/python/resistor-color-trio/1/resistor_color_trio.py
RESISTOR_COLORS = (
    "black",
    "brown",
    "red",
    "orange",
    "yellow",
    "green",
    "blue",
    "violet",
    "grey",
    "white",
)


def value(color):
    return RESISTOR_COLORS.index(color)


def label(colors):
    first_color, second_color, third_color = colors[:3]
    resistance_value = (10 * value(first_color) + value(second_color)) * 10 ** value(
        third_color
    )
    prefix = ""
    if resistance_value >= 1_000_000_000:
        resistance_value /= 1_000_000_000
        prefix = "giga"
    elif resistance_value >= 1_000_000:
        resistance_value /= 1_000_000
        prefix = "mega"
    elif resistance_value >= 1_000:
        resistance_value /= 1_000
        prefix = "kilo"
    return f"{int(resistance_value)} {prefix}ohms"
